pkg/query/storeset.go

```diff
@@ -360,6 +360,12 @@ func externalLabelsFromStore(store *storeRef) string {
 	}
 	sort.Strings(tsdbLabelSetStrings)
 
+	// Append the storeType to the end of list, because the store gateway will be exposed external labels,
+	// we allow the duplicate external labels with different components.
+	if store.storeType != nil {
+		tsdbLabelSetStrings = append(tsdbLabelSetStrings, store.storeType.String())
```

Comment:
Hm, this sounds like a hack, there is no way to check if this guy is a `storeGW` inside checking deduplicated stuff?

Reply:
Sorry, what do you mean?

Reply:
We are adding some artificial labels to the labelset. Why? This sounds wrong? To allow duplicated store gateway labels we just need to avoid detecting this if `store.storeType` is Gateway, right? NOT